pkg/apis/extensions/validation/operatingsystemconfig_test.go

```diff
@@ -301,6 +301,25 @@ var _ = Describe("OperatingSystemConfig validation tests", func() {
 			}))))
 		})
 
+		It("should forbid setting an unknown cgroup driver", func() {
+			oscCopy := osc.DeepCopy()
+			oscCopy.Spec.CRIConfig.CgroupDriver = ptr.To(extensionsv1alpha1.CgroupDriverName("unknown"))
+
+			Expect(ValidateOperatingSystemConfig(oscCopy)).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Type":  Equal(field.ErrorTypeNotSupported),
+				"Field": Equal("spec.criConfig.cgroupDriver"),
+			}))))
+		})
+
+		It("should allow setting a known cgroup driver", func() {
+			for _, driver := range []extensionsv1alpha1.CgroupDriverName{"cgroupfs", "systemd"} {
+				oscCopy := osc.DeepCopy()
+				oscCopy.Spec.CRIConfig.CgroupDriver = ptr.To(driver)
+
+				Expect(ValidateOperatingSystemConfig(oscCopy)).To(BeEmpty())
```

Comment:
Do you want to add context so that it is clear which cgroup driver was used in case of failure (see https://github.com/gardener/gardener/blob/2eb54485231408cbdbabaa49812572a07124364f/test/e2e/shoot/internal/rotation/certificate_authorities.go#L89-L93)?

Reply:
Great suggestion, thanks 👍